### PR TITLE
test-case: add capture pipeline support for runtime pm test

### PR DESCRIPTION
**Pull request by fredoh9**
aplay/arecord can be used by setting -m option. Add scanning dmesg for
error before switching pipelines.

Signed-off-by: Fred Oh <fred.oh@linux.intel.com>

I submitted two commits in in PR#92. This is first part of it.
https://github.com/thesofproject/sof-test/pull/92
For the new test case i will update on PR#92 if I found better solution.

### CODE REVIEW

**Comment by aiChaoSONG** [comment]
@fredoh9  Hi, fred, can you refer to https://github.com/thesofproject/sof-test/pull/102. The runtime pm test should be one test case. with your solution, we have to call the script twice, in fact, we can cover both playback and capture with just one call of this scripts.

**Comment by Bin-QA** [comment]
how about go though the TPLG to remove '-m' option?

File: test-case/check-runtime-pm-status.sh [hunk 1]
```diff
@@ -45,16 +52,31 @@ case $platform in
...
+
 runtime_status="/sys/bus/pci/devices/0000:$(lspci |awk '/[Aa]udio/ {print $1;}')/power/runtime_status"
 
 [[ ! -f $runtime_status ]] && dloge "no runtime_status entry: $runtime_status" && exit 1
 
 dlogc "Runtime status check: cat $runtime_status"
 
 [[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
-func_pipeline_export $tplg "type:playback"
+func_pipeline_export $tplg "type:$test_mode"
```
Comment (Bin-QA) [inline test-case/check-runtime-pm-status.sh]: here you can direct load `func_pipeline_export $tplg` to get full pipeline

File: test-case/check-runtime-pm-status.sh [hunk 2]
```diff
@@ -65,17 +87,17 @@ do
```
Comment (Bin-QA) [inline test-case/check-runtime-pm-status.sh]: you need to add 'type' refer to decide use which command & dummy_file

**Comment by stripes416** [comment]
Good patch, once @aiChaoSONG & @Bin-QA 's suggestions have been made, this will be a good addition!

**Comment by wenqingfu** [comment]
ping @fredoh9  for follow up

**Comment by fredoh9** [comment]
@aiChaoSONG @Bin-QA I rebased and addressed your comments.

File: test-case/check-runtime-pm-status.sh [hunk 3]
```diff
@@ -49,62 +52,80 @@ func_check_dsp_status()
...
+        ;;
+        "capture")
+            cmd=arecord
+            dummy_file=/dev/null
+        ;;
+        *)
+            dloge "Invalid pipeline: $type (allow value : playback, capture)"
+            exit 1
+        ;;
+    esac
```
Comment (Bin-QA) [inline test-case/check-runtime-pm-status.sh]: Almost is good to me   
Can you try the bash skill like this usage:  https://github.com/thesofproject/sof-test/blob/master/test-case/multiple-pipeline-playback.sh#L57   

```
help declare
-A        to make NAMEs associative arrays (if supported)
```

so here you can direct to use like: `cmd=${cmd_lst[$type]}`  
if you worry about miss `keyword`, so you can add check for `cmd` is empty  
you can try this skill, I think it is more useful some simple string transform

**Pull request merged**